feat(chips): Add (select), [color] and dark theme to chips.

topherfangio opened the pull request:
Add new functionality/options to chips for managing selection/color.

*MdChipList Options*

 - `[selectable]` - Programmatically control whether or not the chips
   in the list are capable of being selected.
 - The SPACE key will automatically select the currently focused chip.

*MdChip Options*

 - `[color]` - Programmatically control the selected color of the
   chip.
 - `[selected]` - Programmatically control whether or not the chip
   is selected.
 - `(select)` - Event emitted when the chip is selected.
 - `(deselect)` - Event emitted when the chip is deselected.

Additionally, adds basic support for dark themeed chips using existing
colors from other components in the spec and cleanup demos by using
cards and toolbars like other demos.

References #120.

---
<img width="475" alt="screen shot 2016-12-15 at 2 39 01 pm" src="https://cloud.githubusercontent.com/assets/54370/21241226/41779a00-c2d4-11e6-8dbd-39969ce9647b.png">

jelbourn commented on src/lib/chips/_chips-theme.scss:
```diff
@@ -6,27 +7,36 @@
+  $foreground: map-get($theme, foreground);
+
+  $unselected-background: if($is-dark-theme, md-color($background, card), #e0e0e0);
+  $unselected-foreground: if($is-dark-theme, md-color($foreground, text), rgba(0, 0, 0, 0.87));
```
Why are these not just
```scss
$unselected-background: md-color($background, card);
$unselected-foreground: md-color($foreground, text);
// etc.
```
?

The background and foreground colors are already based on whether it's a light/dark theme. Is there some reason the light theme version of these colors aren't appropriate?

topherfangio commented:
@jelbourn Requested changes made.

**Regarding the left/right arrow keys:** I plan to add that at a later date, but many screen readers have difficulty with left/right which can cause problems (and did so with the md1 version). I'd like to do some testing before we add it so that I can work around any problems.

topherfangio commented:
FYI - I introduced a small bug in my last commit, so I added a new test when I fixed it. Just checks for proper CSS classes when the chip is selected.
